app/models/user.rb

```diff
@@ -19,4 +22,12 @@ def ensure_an_admin_remains
       raise Error.new "Can't delete last user"
     end
   end
+
+  def ensure_not_admin
+    yield unless self.email == 'admin@depot.com'
+  end
+
+  def send_welcome_mail
+    UserMailer.with(user: self).welcome_email.deliver_now
```

Comment:
why deliver_now?
read deliver_now vs deliver_later

Reply:
so that we can send mail on priority. 
but it is recommended to use deliver_later, so that users never wait for mail to deliver
done

Reply:
done